bentoml/_internal/yatai_client/schemas.py

```diff
@@ -0,0 +1,240 @@
+    if not datetime_str:
+        return None
+    return parse(datetime_str)
+
+
+datetime_field_kwargs = dict(metadata={'dataclasses_json': {
+    'encoder': datetime_encoder,
+    'decoder': datetime_decoder,
+    'mm_field': fields.DateTime(format='iso')
+}})
+
+
+@dataclass
+class BaseSchema(DataClassJsonMixin):
+    uid: str = ''
+    created_at: datetime = field(default_factory=datetime.now, **datetime_field_kwargs)
+    updated_at: Optional[datetime] = field(default=None, **datetime_field_kwargs)
+    deleted_at: Optional[datetime] = field(default=None, **datetime_field_kwargs)
+
+
+class BaseListSchema(DataClassJsonMixin):
+    start: int
+    count: int
+    total: int
+
+
+class ResourceType(Enum):
+    USER = 'user'
+    ORG = 'organization'
+    CLUSTER = 'cluster'
+    BENTO = 'bento'
+    BENTO_VERSION = 'bento_version'
+    MODEL = 'model'
+    MODEL_VERSION = 'model_version'
+
+
+@dataclass
+class ResourceSchema(BaseSchema, DataClassJsonMixin):
+    name: str = ''
+    resource_type: ResourceType = ResourceType.USER
+
+
+@dataclass
+class LabelItemSchema(DataClassJsonMixin):
+    key: str = ''
+    value: str = ''
+
+
+@dataclass
+class UserSchema(ResourceSchema):
+    name: str = ''
+    email: str = ''
+    first_name: str = ''
+    last_name: str = ''
+
+    def get_name(self) -> str:
+        if not self.first_name and not self.last_name:
+            return self.name
+        return f'{self.first_name} {self.last_name}'.strip()
+
+
+@dataclass
+class OrganizationSchema(ResourceSchema, DataClassJsonMixin):
+    description: str = ''
+
+
+@dataclass
+class OrganizationListSchema(BaseListSchema):
+    items: List[OrganizationSchema]
+
+
+@dataclass
+class ClusterSchema(ResourceSchema):
+    description: str = None
+
+
+@dataclass
+class BentoSchema(ResourceSchema):
+    description: str = None
+    latest_version: Optional["BentoVersionSchema"] = None
+
+
+@dataclass
+class CreateBentoSchema(DataClassJsonMixin):
+    name: str = ''
+    description: str = ''
+
+
+class BentoVersionImageBuildStatus(Enum):
+    PENDING = 'pending'
+    BUILDING = 'building'
+    SUCCESS = 'success'
+    FAILED = 'failed'
+
+
+class BentoVersionUploadStatus(Enum):
+    PENDING = 'pending'
+    BUILDING = 'uploading'
+    SUCCESS = 'success'
+    FAILED = 'failed'
+
+
+@dataclass
+class BentoVersionApiSchema(DataClassJsonMixin):
+    route: str = ''
+    doc: str = ''
+    input: str = ''
+    output: str = ''
+
+
+@dataclass
+class BentoVersionManifestSchema(DataClassJsonMixin):
+    service: str = ''
+    bentoml_version: str = ''
+    apis: Dict[str, BentoVersionApiSchema] = field(default_factory=dict)
+    models: List[str] = field(default_factory=list)
+
+
+@dataclass
+class BentoVersionSchema(ResourceSchema):
+    description: str = ''
+    version: str = ''
+    image_build_status: BentoVersionImageBuildStatus = ''
+    upload_status: BentoVersionUploadStatus = ''
+    upload_finished_reason: str = ''
+    presigned_s3_url: str = ''
```

Comment:
could we change the API to just `presigned_download_url`? that way we don't need to change this API when we start adding support for other cloud blob storage services (azure blob storage, google cloud storage etc)